pyi.py

```diff
@@ -925,11 +890,12 @@ def visit_Assign(self, node: ast.Assign) -> None:
                 )
             return
 
-        if self._Y015_violation_detected(node):
-            return self._Y015_error(node)
-
         if not is_special_assignment:
             self._check_for_type_aliases(node, target, assignment)
+            if isinstance(
+                assignment, (ast.Dict, ast.List, ast.Tuple, ast.Set, ast.JoinedStr)
```

Comment:
Why doesn't this use `_is_valid_stub_default`? There are a lot of other expression nodes.

Reply:
Yup. But for assignments, we want to allow much more than just the things we allow in default values for parameters. All of the following are valid in a stub:

```python
T = TypeVar("T") # ast.Call

def foo(): ...
bar = foo # ast.Name

class Spam:
    def eggs(self): ...
ham = Spam.eggs  # ast.Attribute

baz = Literal["foo"]  # ast.Subscript — this should trigger Y026 but not Y015
```

Reply:
I see, but this still allows a bunch of nodes that have little conceivable use, like `Lambda` and `SetComp`. Might be better to go with an allowlist instead of a blocklist.

Reply:
Hmm, fair enough, I'll rework.